src/Compilers/CSharp/Test/Semantic/Semantics/RecordTests.cs

```diff
@@ -1514,6 +1697,91 @@ public static void Main()
             var xId = withExpr.DescendantNodes().Single(id => id.ToString() == "X");
             var symbolInfo = model.GetSymbolInfo(xId);
             Assert.True(x.ISymbol.Equals(symbolInfo.Symbol));
+
+            comp.VerifyOperationTree(withExpr, @"
+IWithExpressionOperation (OperationKind.WithExpression, Type: C) (Syntax: 'c with { X = 2 }')
+  Value:
+    ILocalReferenceOperation: c (OperationKind.LocalReference, Type: C) (Syntax: 'c')
+  CloneMethod: C C.Clone()
+  Initializer:
+    IObjectOrCollectionInitializerOperation (OperationKind.ObjectOrCollectionInitializer, Type: C) (Syntax: '{ X = 2 }')
+      Initializers(1):
+          ISimpleAssignmentOperation (OperationKind.SimpleAssignment, Type: System.Int32) (Syntax: 'X = 2')
+            Left:
+              IPropertyReferenceOperation: System.Int32 C.X { get; init; } (OperationKind.PropertyReference, Type: System.Int32) (Syntax: 'X')
+                Instance Receiver:
+                  IInstanceReferenceOperation (ReferenceKind: ImplicitReceiver) (OperationKind.InstanceReference, Type: C, IsImplicit) (Syntax: 'X')
```

Comment:
>IInstanceReferenceOperation (ReferenceKind: ImplicitReceiver) [](start = 18, length = 61)

Consider adjusting the comment for the node and the ReferenceKind to include information about this usage